sky/serve/load_balancer.py

```diff
@@ -18,25 +21,135 @@
 
 logger = sky_logging.init_logger(__name__)
 
+# TODO(ejj): Major things that may be needed before merging Envoy Support.
+# This comment should go away before merging.
+#
+# - Automated Unit/Smoke Testing: Obviously this needs to be implemented before
+# merging.  For now the code was tested by hand.
+#
+# - Hitless Configuration Changes: When configured with a yaml file, Envoy must
+# be restarted to take up configuration changes.  When done naively (as the
+# PR currently does), restarting will cause all existing connections to be
+# dropped, forcing clients to retry later.  As written, this means that any
+# time a replica is added or removed, connections to all replicas will be
+# impacted.  There are two ways to deal with this:
+#   - Hot Restart:  Envoy has a mechanism for "hot restart".   When used, a new
+#   Envoy boots and begins handling connections with the new coniguration while
+#   the old one drains.
```

Comment:
yeah unfortunately this is a bit more difficult in Envoy than in ngninx, largely because Envoy is usually configured through xDS (their remote API) rather than config files like this.  hot-restarter.py is exactly what I was planning to use initially, though eventually, particularly when we've got multiple of these things in different regions, xDS will solve this problem entirely.

Reply:
latest version of the PR fixes this.